tests/renderer/components/dialog-bisect-spec.tsx

```diff
@@ -79,7 +79,7 @@ describe('BisectDialog component', () => {
       const wrapper = shallow(<BisectDialog appState={store as any} />);
       const instance: any = wrapper.instance() as any;
 
-      expect(instance.state.startIndex).toBe(NUM_VERSIONS);
+      expect(instance.state.startIndex).toBe(NUM_VERSIONS - 1);
```

Comment:
Could we extend this test to have test coverage for both cases: > 10 and <= 10?

I think we could change the test to be `describe.each([8, 15])('BisectDialog component', (numVersions) => {` and adjust accordingly.

This `expect` could then be changed to: `expect(instance.state.startIndex).toBeLessThan(numVersions);`, which would ensure we don't regress where the `startIndex` is larger than the number of versions. You could then add another `expect` to check the logic around setting the start index at 10 if there's more than 10 versions, like: `expect(instance.state.startIndex).toBe(numVersions > 10 ? 10 : numVersions - 1);`

I think that would give us robust test coverage for this issue. 🙂

Reply:
Good idea!